### DISCUSSION TITLE
안녕하세요. Part3의 Chapter4에서 NodePort 관련 질문이 있습니다

### DISCUSSION THREAD

**Discussion by hanki-cho** [yoonjeong-kwon/fastcampus-kubernetes#8] [Q&A] [answered]
안녕하세요. Part3의 Chapter4에서 NodePort 관련 질문이 있습니다

전체적인 구성도 중에서
ClusterIP로 선언된 payment service 에서 클라이언트의 checkout 요청에 응답을 주는 것으로 되어 있는데요
외부와 연결되지 않은 payment가 어떻게 클라이언트에 응답을 줄 수 있는 건가요?

혹시 실제로는 payment 는 order가 요청하는 receipt의 응답을 주는거고,
order가 응답을 받아서 클라이언트의 checkout 응답을 주는게 맞는 걸까요?

아니면 외부와 연결되어 있지 않은 ClusterIP도 응답을 주는 것은 가능한건가요?

**Comment by yoonjeong-kwon** [comment, marked as answer]
@decafandmac 님, 안녕하세요. 

선을 단순화해서 그리다보니 payment service가 클라이언트로 직접 응답을 보내는 것처럼 표현하였습니다. 이부분이 혼선을 드릴 수 있었겠네요. 😅 말씀하신대로 실제로는 payment -> order로 응답을 전송하고 order가 클라이언트로 응답하게 됩니다.

좋은 피드백 감사합니다!
